include/seqan3/alignment/matrix/detail/alignment_matrix_column_major_range.hpp

```diff
@@ -0,0 +1,404 @@
+        ~iterator_type() = default;                                              //!< Defaulted.
+
+        /*!\brief Construction from an instance of the derived type.
+         * \param me     An instance of the derived type.
+         */
+        constexpr iterator_type(derived_t & me) :
+            me_ptr{&me},
+            column_index{0}
+        {}
+        //!\}
+
+        //!\brief Returns the current alignment column.
+        constexpr reference operator*() const noexcept
+        {
+            static_assert(std::ConvertibleTo<decltype(me_ptr->initialise_column(column_index)), reference>,
+                          "The returned type of initialise_column must be convertible to the reference type.");
+            return me_ptr->initialise_column(column_index);
+        }
+
+        //!\brief Increments by one.
+        constexpr iterator_type & operator++() noexcept
+        {
+            ++column_index;
+            return *this;
+        }
+
+        //!\brief Increments by one.
+        constexpr void operator++(int) noexcept
+        {
+            ++(*this);
+        }
+
+        /*!\name Comparison operators
+         * \{
+         */
+        //!\brief Tests whether the behind-the-end column was reached.
+        constexpr bool operator==(std::ranges::default_sentinel_t const &) const noexcept
+        {
+            return column_index == me_ptr->cols;
+        }
+
+        //!\copydoc operator==
+        friend constexpr bool operator==(std::ranges::default_sentinel_t const & lhs, iterator_type const & rhs)
+            noexcept
+        {
+            return rhs == lhs;
+        }
+
+        //!\copydoc operator==
+        constexpr bool operator!=(std::ranges::default_sentinel_t const & rhs) const noexcept
+        {
+            return !(*this == rhs);
+        }
+
+        //!\copydoc operator==
+        friend constexpr bool operator!=(std::ranges::default_sentinel_t const & lhs, iterator_type const & rhs)
+            noexcept
+        {
+            return rhs != lhs;
+        }
+        //!\}
+
+    private:
+
+        //!\brief Pointer to the derived type.
+        derived_t * me_ptr;
+        //!\brief The current column index.
+        size_t column_index{};
+    };
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr alignment_matrix_column_major_range() = default; //!< Defaulted.
+    constexpr alignment_matrix_column_major_range(alignment_matrix_column_major_range const &) = default; //!< Defaulted.
+    constexpr alignment_matrix_column_major_range(alignment_matrix_column_major_range &&) = default; //!< Defaulted.
+    constexpr alignment_matrix_column_major_range & operator=(alignment_matrix_column_major_range const &) = default; //!< Defaulted.
+    constexpr alignment_matrix_column_major_range & operator=(alignment_matrix_column_major_range &&) = default; //!< Defaulted.
+    ~alignment_matrix_column_major_range() = default; //!< Defaulted.
```

Comment:
align the `=` operators, please

Reply:
What about `noexcept`?

Reply:
Regarding `noexcept`: Even if it is automatically noexcept, it would be helpful to see the keyword in the code and the documentation. 

Regarding the alignment: This is a PR and the name is therefore not likely to change again. I just wanted to point out that you start introducing a second code style into the library, because I usually see the rule-of-six aligned.